tests/test_report_reader.cpp

```diff
@@ -23,23 +30,89 @@ TEST_CASE("SpikeReader", "[base]") {
+
+    // Negative ids
+    REQUIRE_THROWS(pop.get(Selection({{-1, 1}})));
+
+    // Times out of range
+    REQUIRE(pop.get(Selection({{1, 2}}), 100., 101.).ids == DataFrame<NodeID>::DataType());
+    testTimes(pop.get(Selection({{1, 2}}), 100., 101.).times, 0.0, 0.0, 0);
+    REQUIRE(pop.get(Selection({{1, 2}}), 100., 101.).data == std::vector<std::vector<float>>({}));
+
+    // Inverted times
+    REQUIRE(pop.get(Selection({{1, 2}}), 0.2, 0.1).ids == DataFrame<NodeID>::DataType({1}));
+    testTimes(pop.get(Selection({{1, 2}}), 0.2, 0.1).times, 0.2, 0.1, 1);
+    REQUIRE(pop.get(Selection({{1, 2}}), 0.2, 0.1).data ==
+            std::vector<std::vector<float>>({{1.2}}));
+
+    // Negatives times
+    REQUIRE(pop.get(Selection({{1, 2}}), -1., -2.).ids == DataFrame<NodeID>::DataType({1}));
+    testTimes(pop.get(Selection({{1, 2}}), -1., -2.).times, 0.0, 0.1, 10);
+    REQUIRE(pop.get(Selection({{1, 2}}), -1., -2.).data ==
+            std::vector<std::vector<float>>(
+                {{1.0}, {1.1}, {1.2}, {1.3}, {1.4}, {1.5}, {1.6}, {1.7}, {1.8}, {1.9}}));
+}
+
+TEST_CASE("SomaReportReader", "[base]") {
+    const SomaReportReader reader("./data/somas.h5");
+
+    REQUIRE(reader.getPopulationsNames() == std::vector<std::string>{"All", "soma1", "soma2"});
+
+    auto pop = reader.openPopulation("All");
+
+    REQUIRE(pop.getTimes() == std::make_tuple(0., 1., 0.1));
+
+    REQUIRE(pop.getTimeUnits() == "ms");
+
+    REQUIRE(pop.getDataUnits() == "mV");
+
+    REQUIRE(pop.getSorted());
+
+    auto data = pop.get(Selection({{3, 5}}), 0.2, 0.5);
+    REQUIRE(data.ids == DataFrame<NodeID>::DataType{{3, 4}});
+    testTimes(data.times, 0.2, 0.1, 4);
+    REQUIRE(data.data == std::vector<std::vector<float>>{
+                             {{3.2f, 4.2f}, {3.3f, 4.3f}, {3.4f, 4.4f}, {3.5f, 4.5f}}});
+}
+
+TEST_CASE("ElementReportReader", "[base]") {
+    const ElementReportReader reader("./data/elements.h5");
+
+    REQUIRE(reader.getPopulationsNames() ==
+            std::vector<std::string>{"All", "element1", "element42"});
+
+    auto pop = reader.openPopulation("All");
 
-    REQUIRE(reader.openPopulation("All").getTimeUnits() == "ms");
+    REQUIRE(pop.getTimes() == std::make_tuple(0., 4., 0.2));
 
-    REQUIRE(reader.openPopulation("All").getDataUnits() == "mV");
+    REQUIRE(pop.getTimeUnits() == "ms");
 
-    REQUIRE(reader.openPopulation("All").getSorted());
+    REQUIRE(pop.getDataUnits() == "mV");
 
-    REQUIRE(reader.openPopulation("All").get(Selection({{3, 5}}), 0.2, 0.5).ids ==
-            DataFrame<NodeID>::DataType{{3, 4}});
+    REQUIRE(pop.getSorted());
 
-    REQUIRE(
-        reader.openPopulation("All").get(Selection({{3, 5}}), 0.2, 0.5).data ==
-        std::vector<std::vector<float>>{{{3.2f, 4.2f}, {3.3f, 4.3f}, {3.4f, 4.4f}, {3.5f, 4.5f}}});
+    auto data = pop.get(Selection({{3, 5}}), 0.2, 0.5);
+    REQUIRE(data.ids ==
+            DataFrame<std::pair<NodeID, uint32_t>>::DataType{
```

Comment:
Could it be beneficial to have a :
```
using ElementID = uint32_t;
```
?

Reply:
Yes ;)

Reply:
Cool. 
I was puzzled by the `uint32_t` and I had to check the specs again ...

Reply:
ping !